src/moscot/problems/cross_modality/_translation.py

```diff
@@ -0,0 +1,217 @@
+
+    Parameters
+    ----------
+    adata_src
+        Instance of :class:`anndata.AnnData` containing the source data.
+    adata_tgt
+        Instance of :class:`anndata.AnnData` containing the target data.
+    """
+
+    def __init__(self, adata_src: AnnData, adata_tgt: AnnData, **kwargs: Any):
+        super().__init__(adata_src, **kwargs)
+        self._adata_tgt = adata_tgt
+        self.filtered_vars: Optional[Sequence[str]] = None
+
+    def _create_policy(  # type: ignore[override]
+        self,
+        policy: Literal["external_star"] = "external_star",
+        key: Optional[str] = None,
+        **kwargs: Any,
+    ) -> Union[DummyPolicy, ExternalStarPolicy[K]]:
+        """Private class to create DummyPolicy if no batches are present in the spatial anndata."""
+        del policy
+        if key is None:
+            return DummyPolicy(self.adata, **kwargs)
+        return ExternalStarPolicy(self.adata, key=key, **kwargs)
+    
+    def _create_problem(
+        self,
+        src: K,
+        tgt: K,
+        src_mask: ArrayLike,
+        tgt_mask: ArrayLike,
+        **kwargs: Any,
+    ) -> OTProblem:
+        return self._base_problem_type(
+            adata=self.adata_src,
+            adata_tgt=self.adata_tgt,
+            src_obs_mask=src_mask,
+            tgt_obs_mask=None,
+            src_key=src,
+            tgt_key=tgt,
+            **kwargs,
+        )
+
+    @d.dedent
+    def prepare(
+        self,
+        src_attr: Str_Dict_t,
+        tgt_attr: Str_Dict_t,
+        var_names: Optional[Sequence[Any]] = None,
+        joint_attr: Optional[Union[str, Mapping[str, Any]]] = None,
+        cost: Union[
+            Literal["sq_euclidean", "cosine", "bures", "unbalanced_bures"],
+            Mapping[str, Literal["sq_euclidean", "cosine", "bures", "unbalanced_bures"]],
+        ] = "sq_euclidean",
+        a: Optional[str] = None,
+        b: Optional[str] = None,
+        **kwargs: Any,
+    ) -> "TranslationProblem[K]":
+        """
+        Prepare the :class:`moscot.problems.cross_modality.TranslationProblem`.
+
+        This method prepares the data to be passed to the optimal transport solver.
+
+        Parameters
+        ----------
+        
+
+        Returns
+        -------
+        :class:`moscot.problems.cross_modality.TranslationProblem`.
+
+        """
+        self._src_attr = src_attr['key']
+        self._tgt_attr = tgt_attr['key']
+
+        x = {"attr": "obsm", "key": src_attr} if isinstance(src_attr, str) else src_attr
+        y = {"attr": "obsm", "key": tgt_attr} if isinstance(tgt_attr, str) else tgt_attr
+        self.filtered_vars = var_names
+        if self.filtered_vars is not None:
+            xy, kwargs = handle_joint_attr(joint_attr, kwargs)
+        else:
+            xy = None
+        #xy, kwargs = handle_joint_attr(joint_attr, kwargs)
+        xy, x, y = handle_cost(xy=xy, x=x, y=y, cost=cost)
+        if xy is not None:
+            kwargs["xy"] = xy
```

Comment:
Included `filtered_vars` to run the notebook. Otherwise I got the following error message:
![image](https://user-images.githubusercontent.com/114418967/228082545-64895fc3-7dce-4365-b358-d865b8c09b28.png)

Reply:
can you please paste the complete traceback of the error message?